libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -500,6 +501,32 @@ export class WorkspaceInstallManager extends BaseInstallManager {
       await doInstallInternalAsync(this.options);
     }
 
+    // if usePnpmSyncForInjectedDependencies is true
+    // the pnpm-sync will generate the pnpm-sync.json based on lockfile
+    if (this.rushConfiguration.packageManager === 'pnpm' && experiments?.usePnpmSyncForInjectedDependencies) {
+      const pnpmLockfilePath: string = this.rushConfiguration.tempShrinkwrapFilename;
+      const pnpmStorePath: string = `${this.rushConfiguration.commonTempFolder}/node_modules/.pnpm`;
+      await pnpmSyncPrepareAsync({
+        lockfilePath: pnpmLockfilePath,
+        storePath: pnpmStorePath,
+        readWantedLockfile: async (pnpmLockFolderPath) => {
+          const wantedPnpmLockfile: PnpmShrinkwrapFile | undefined = await PnpmShrinkwrapFile.loadFromFile(
+            pnpmLockFolderPath + 'pnpm-lock.yaml',
```

Comment:
```suggestion
            pnpmLockFolderPath + RushConstants.projectShrinkwrapFilename,
```

Is `pnpmLockFolderPath` guaranteed to have a trailing slash?

Reply:
I am still thinking on this
Maybe we can refactor the `readWantedLockfile `  function, to take the pnpmLockFilePath as input

Originally, `readWantedLockfile` is from [PNPM](https://github.com/pnpm/pnpm/blob/9f8948c24a51bf3ba02cb9981b300671a5304106/lockfile/lockfile-file/README.md?plain=1#L17)
In their implemention, it takes the pnpmLockFolderPath

Reply:
To answer your question, yes, right now, it is guaranteed to have a trailing slash https://github.com/tiktok/pnpm-sync/blob/main/packages/pnpm-sync-lib/src/pnpmSyncPrepare.ts#L52-L59

Reply:
If/when you change it, you should change the function name to not match the name from pnpm's API.

But yes, I think the function should be changed to take the full lockfile path.

Reply:
Yeah, I rename the function to avoid the confusion. https://github.com/tiktok/pnpm-sync/pull/11